### ISSUE TITLE
Derive new zettel from visually selected or motion

### ISSUE THREAD

**Issue by ihsanturk** [ihsanturk/neuron.vim#15]

**Comment by ciderale** [comment]
That's great to see progress on this feature. It was something what I was missing most at the moment.
I also experimented with vimscript, but I see @TyberiusPrime was quicker :-) 

My script look quite similar, but my approach was aiming for one additional thing.
Namely, replacing the visual selection with a reference to the newly created zettel.
I'm not sure, if that is desired, but I find it quite handy to split a zettel into mutliple ones.
It's a bit like an "extract method" refactoring in programming parlance. What do you think about this?

**Comment by TyberiusPrime** [comment]
Personally, I think that to be a good idea - at least optionally. It would also give a very nice, write title, select, new zettel (already linked), write workflow instead of my current new zettel/write title/write, remember to go back and insert last zettel Id, which is prone to breakage if I have to reference other zettels in-between.

**Comment by ciderale** [comment]
You are right, maybe that step should be optional. 

Another question is where should the user end up at the end, (a) on the new zettel, ready for editing the zettel or (b) on the original zettel where the reference is added. If the user aims for splitting a zettel that has grown to big, option (b) might be quite handy. I don't have a preference on this and it's simple to jump back and forth anyway.

I add the core bits here in case that @ihsanturk decides that this is a valuable thing to add. This would be option (a). Since it's the first time a write vim-script, there are probably many things to improve upon. 
```vim-script
func! neuron#extract_and_edit_new_zettel()
  let content = neuron#visualselection()
  exec 'e '. system('neuron -d '.shellescape(g:zkdir).' new "'.content.'"')
  exec 'bd'
  call neuron#refresh_cache()
  exec 'norm"_gvd'
  call neuron#insert_zettel_last()
  exec 'w!'
  " jumping to the new zettel maybe an optional step
  call neuron#edit_zettel_under_cursor()
endf
```

**Comment by TyberiusPrime** [comment]
Might want to shell escape that content as well? What happens if there is a newline in there? (My vimscript is very basic, just worried)

**Comment by ciderale** [comment]
Good catch. I think newline works okay (I think I tested that), but quotes may indeed cause problems.
